File: samples/Go/trabb-pardo-knuth-algorithm.go
package main

import (
    "fmt"
    "log"
    "math"
)

func main() {
    // prompt
    fmt.Print("Enter 11 numbers: ")
    // accept sequence
    var s [11]float64
    for i := 0; i < 11; {
        if _, err := fmt.Scanf("%f", &s[i]); err == nil {
            i++
        }
    }
    // reverse sequence
    for i, item := range s[:5] {
        s[i], s[10-i] = s[10-i], item
    }
    // iterate
    for _, item := range s {
        if result, overflow := f(item); overflow {
            // send alerts to stderr
            log.Printf("f(%g) overflow", item)
        } else {
            // send normal results to stdout
            fmt.Printf("f(%g) = %g\n", item, result)
        }
    }
}

func f(x float64) (float64, bool) {
    result := math.Pow(math.Abs(x), .5) + 5*x*x*x
    return result, result > 400
}

File: samples/Haskell/trabb-pardo-knuth-algorithm.hs
import Control.Monad (replicateM, mapM_)

f x = (abs x) ** 0.5 + 5 * x ** 3

main = do
        putStrLn "Enter 11 numbers for evaluation"
        x <- replicateM 11 $ readLn
        mapM_ ((\x -> if x > 400
                      then putStrLn "OVERFLOW"
                      else print x). f) $ reverse x
